src/particle_filter.h

```diff
@@ -1,54 +1,52 @@
 #ifndef ED_LOCALIZATION_PARTICLE_FILTER_H_
 #define ED_LOCALIZATION_PARTICLE_FILTER_H_
 
+#include "kdtree.h"
+#include "transform.h"
+
 #include <geolib/datatypes.h>
 
+#include <tue/config/configuration.h>
+
 #include <vector>
 
 // ----------------------------------------------------------------------------------------------------
 
-class Transform
+struct Sample
 {
+    Sample() {}
 
-public:
-
-    inline const geo::Transform2& matrix() const { return pose_; }
-    inline double rotation() const { return rotation_; }
-    inline const geo::Vec2& translation() const { return pose_.t; }
-
-    inline void setTranslation(const geo::Vec2& v) { pose_.t = v; }
-    inline void set(const geo::Transform2& p)
-    {
-        pose_ = p;
-        rotation_ = p.rotation();
-    }
-
-    inline void setRotation(double rot)
+    Sample(const geo::Transform2& t)
     {
-        pose_.setRotation(rot);
-        rotation_ = rot;
+        pose.set(t);
```

Comment:
So what does this mean for the weight? Is that undefined?

Reply:
Yes, it was. I have initialized it to 0.